src/acl.c

```diff
@@ -2790,6 +2790,14 @@ void aclCommand(client *c) {
             return;
         }
 
+        if (doesCommandHaveKeys(cmd) &&
+            ((cmd->arity > 0 && cmd->arity != c->argc-3) ||
+            (c->argc-3 < -cmd->arity)))
+        {
+            rejectCommandFormat(c,"wrong number of arguments for '%s' command", cmd->fullname);
```

Comment:
```suggestion
            addReplyErrorFormat(c,"wrong number of arguments for '%s' command", cmd->fullname);
```
`rejectCommandFormat` is reserved for processCommand, in this case the command (dryrun) "failed" it was not "rejected"